embryo/segment_2D_0.py

```diff
@@ -91,23 +84,42 @@ def plot_comparison(plot1, plot2, title1, title2):
 
 # %%
 # Deep dive into local thresholding:
-local_thresh_1 = threshold_local(frame, block_size=35, offset=10)
+local_thresh_1 = ski.filters.threshold_local(frame, block_size=35, offset=10)
 mask_1 = frame > local_thresh_1
-local_thresh_2 = threshold_local(frame, block_size=35)
+local_thresh_2 = ski.filters.threshold_local(frame, block_size=35)
 mask_2 = frame > local_thresh_2
 plot_comparison(mask_1, mask_2, "Offset = 10", "Without Offset")
 # Increasing the offset value yields more details of the image
 
 # %%
-local_thresh_1 = threshold_local(frame, block_size=25, offset=10)
+local_thresh_1 = ski.filters.threshold_local(frame, block_size=25, offset=10)
 mask_1 = frame > local_thresh_1
-local_thresh_2 = threshold_local(frame, block_size=35, offset=10)
+local_thresh_2 = ski.filters.threshold_local(frame, block_size=35, offset=10)
 mask_2 = frame > local_thresh_2
 plot_comparison(mask_1, mask_2, "block_size = 25", "block_size = 35")
 # The nuclei are more defined when block_size is set to a higher value
 
 # %%
 # Try Multi-Otsu thresholding
-thresholds = threshold_multiotsu(frame, classes=3)
+thresholds = ski.filters.threshold_multiotsu(frame, classes=3)
 regions = np.digitize(frame, bins=thresholds)
 plot_comparison(frame, regions, "Original", "Multi-Otsu thresholding")
+
+# %%
+# Plot image histogram
+fig, ax = plt.subplots(ncols=1)
+
+ax.hist(frame.ravel(), bins=25)
+ax.set_title('Histogram')
+
+# %%
+# Enhance contrast by equalizing histogram
+enhanced_image = ski.exposure.equalize_hist(frame)
```

Comment:
Nice! Along the same line, I tried adaptive histogram equalization (AHE); then I created markers (as in the [tutorial](https://scikit-image.org/docs/stable/user_guide/tutorial_segmentation.html)) for use in watershed segmentation. Here's the code:

```py
ahe_img = ski.exposure.equalize_adapthist(frame)

markers = np.zeros_like(frame)
markers[ahe_img < 0.1] = 1
markers[ahe_img > 0.4] = 2

segmentation = ski.segmentation.watershed(ahe_img, markers)
labels = ski.measure.label(segmentation)

fig, ax = plt.subplots()
ax.imshow(labels[1000:1250, 1000:1250], cmap='flag')
fig.savefig(fname="watershed.png")
```

![watershed](https://github.com/mkcor/draft-notebooks/assets/2227806/b0b6bc8d-b2c3-496d-a162-927375c307f6)

I'm visualizing a small window of the XY slice, as if we were zooming in (PS: I chose this colormap just for fun!). Visualizing the entire image, you can see that we're missing nuclei located in the darker regions... Maybe we can get some improvement by tweaking the `clip_limit` parameter of `ski.exposure.equalize_adapthist()`?

Reply:
> XY slice

My bad! It's not a *slice* but a *projection* of the entire embryo (at this time point) onto the XY plane. Otherwise, we would have a *stack* of N (2D) XY files, not a single `..._xyProjection.corrected.shifted.klb` file, when we [download the data](https://github.com/mkcor/draft-notebooks/wiki/Mouse-Embryo-Dataset#load-dataset).